pkg/storage/ingest/reader.go

```diff
@@ -218,8 +231,7 @@ func (r *PartitionReader) stopDependencies() error {
 
 func (r *PartitionReader) run(ctx context.Context) error {
 	for ctx.Err() == nil {
-		fetches := r.pollFetches(ctx)
-		err := r.processFetches(ctx, fetches, r.metrics.receiveDelayWhenRunning)
+		err := r.processNextFetches(ctx, r.metrics.receiveDelayWhenRunning)
```

Comment:
Now that we have a separate pattern for calling pollFetches, you no longer need to pass the metrics to `processNextFetchets`.

So you can just do:

```golang
r.metrics.receiveDelayWhenRunning.Observe(now.Sub(record.Timestamp).Seconds())
```

instead of:

```golang
delayObserver.Observe(now.Sub(record.Timestamp).Seconds())
```

Reply:
ah, right. good point 👍

Reply:
actually, I'm not sure I get it - we still have two flows for calling `processNextFetches` - one passing `receiveDelayWhenRunning` and the other `receiveDelayWhenStarting`.

Reply:
Good point, I missed this. I'm not in love with this pattern and I've got an idea on how to simplify it but let's keep the train moving we can do that later.